### ISSUE TITLE
Not all contacts are shown when inviting friends

### ISSUE THREAD

**Issue by MaartenDeen** [signalapp/Signal-Android#6033]
### Bug description
When I use the menu option to invite friends, some contacts are not listed.

### Device info
**Device:** Wiko Rainbow Jam
**Android version:** 5.1
**Signal version:** 3.26.2

**Comment by 2-4601** [comment]
Are the contacts that don't show up stored in SIM card?

**Comment by MaartenDeen** [comment]
All my contacts are google contacts. When I look into the details of the contact, at the bottom there is a line called "Signal contact" and I can see details there.

Hmm, does that maybe mean that that contact is already using signal and is signal quietly ignoring that contact?

**Comment by niffrig** [comment]
TL;DR I'm not sure if this is signal specific or if it's directly solvable through the application.  I see similar strangeness throughout the Android ecosystem WRT contacts.

TL: 
This is a persistent problem for me with Android/Google Contacts/Google Plus and the various applications that I use that rely on those stores for contacts.  I have contacts that show valid phone numbers when viewing the contact but when receiving an SMS through Hangouts/Messenger/Signal the contact record is not resolved and I get an unknown Avatar.  Usually editing the contact resolves the issue... however that's tedious.  This likely manually fixes some otherwise broken link between databases that are obscured to the user.

**Comment by FeuRenard** [comment]
#5856 describes that only numbers are shown of contacts without Signal registration. So if one contact with two numbers has one of them Signal registered, both numbers won't appear when you attempt to invite people.

**Comment by FeuRenard** [comment]
Please check, whether this issue is fixed for you by dc18f73 / #6475 (in 4.6.0). Should be the case 😉 

Personally I consider your desired behaviour as an issue. If one of my contacts has one of her/his numbers registered with Signal, why should I invite another number of her/him?

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
